internal/credit/postgres_connector/grant.go

```diff
@@ -273,31 +260,25 @@ func (c *PostgresConnector) GetGrant(ctx context.Context, namespace string, id s
 // Reset resets the ledger for the subject.
 // Rolls over grants with rollover configuration.
 func (c *PostgresConnector) Reset(ctx context.Context, namespace string, reset credit_model.Reset) (credit_model.Reset, []credit_model.Grant, error) {
-	_, err := c.checkHighWatermark(ctx, namespace, reset.Subject, reset.EffectiveAt)
-	if err != nil {
-		return credit_model.Reset{}, nil, err
-	}
+	var rolloverGrants []credit_model.Grant
 
-	// Start a transaction
-	tx, err := c.db.Tx(ctx)
-	if err != nil {
-		return credit_model.Reset{}, nil, err
-	}
+	reset, err := syncronizedTransaction(ctx, c, namespace, reset.Subject, func(tx *db.Tx, highWatermark time.Time) (*R, error) {
+		var err error
+		// or we can wrap this into a single struct or make syncronized transaction to return err only
+		// and always use variable from the outer scope
+		reset, rolloverGrants, err = c.resetWithTransaction(tx, ctx, namespace, reset)
```

Comment:
reset could accept the highWatermark from the args of the callback as an optimization.